api.js

```diff
@@ -44,19 +45,36 @@ const pullDeleting = (prop, obj) => {
   return v;
 };
 
-const makeBodiedRequest = (req, bodyProp, body, auth) =>
-  extractBody(
-    splitBodyProps(body).then(({ left: arr, right: nonArr }) =>
-      // console.error(arr, nonArr),
-      Object.keys(arr).reduce(
-        (req, k) => (applyArrayParams(k, arr[k], req), req),
-        req.use(log).field({ ...fixParams(nonArr)(pullDeleting('asPropOf', nonArr)), ...auth }),
-      ),
-    ),
-  )(bodyProp);
+const sleep = (exports.sleep = (seconds = 0.75) => {
+  const ms = seconds * 1000;
+  const start = Date.now();
+  while (Date.now() - start < ms) {}
+});
+
+// use to avoid tripping the discourse request limiter
+const sleepAsync = seconds =>
+  new Promise(res => {
+    sleep(seconds);
+    res();
+  });
+
+const defaultSleepSeconds = 1;
+
+const makeBodiedRequest = function makeBodiedRequest(req, bodyProp, body, auth) {
+  return sleepAsync(defaultSleepSeconds).then(() => {
+    const string = qs.stringify(body);
```

Comment:
Still really surprised this works! Have you tried it out with the dispute-tools already?

Reply:
I had not! I just checked and it does work--but we definitely don't want to sleep before calling discourse.